src/memory-analyzer/memory_analyzer_parse_options.cpp

```diff
@@ -0,0 +1,183 @@
+  }
+
+  if(!core_file && !breakpoint)
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "need to provide either core-file or breakpoint for gdb",
+      "--core-file/--breakpoint");
+  }
+
+  const bool output_file = cmdline.isset("output-file");
+  const bool symtab_snapshot = cmdline.isset("symtab-snapshot");
+
+  if(symtab_snapshot && output_file)
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "printing to a file is not supported for symbol table snapshot output",
+      "--symtab-snapshot");
+  }
+
+  register_language(new_ansi_c_language);
+
+  std::string binary = cmdline.args.front();
+
+  const std::string symbol_list(cmdline.get_value("symbols"));
+  std::vector<std::string> result;
+  split_string(symbol_list, ',', result, true, true);
+
+  auto opt = read_goto_binary(binary, ui_message_handler);
+
+  if(!opt.has_value())
+  {
+    throw deserialization_exceptiont(
+      "cannot read goto binary `" + binary + "'");
+  }
+
+  const goto_modelt goto_model(std::move(opt.value()));
+
+  gdb_value_extractort gdb_value_extractor(
+    goto_model.symbol_table, binary.c_str());
+  gdb_value_extractor.create_gdb_process();
+
+  if(core_file)
+  {
+    std::string core_file = cmdline.get_value("core-file");
+    gdb_value_extractor.run_gdb_from_core(core_file);
+  }
+  else if(breakpoint)
+  {
+    std::string breakpoint = cmdline.get_value("breakpoint");
+    gdb_value_extractor.run_gdb_to_breakpoint(breakpoint);
+  }
+  gdb_value_extractor.analyze_symbols(result);
+
+  std::ofstream file;
+
+  if(output_file)
+  {
+    file.open(cmdline.get_value("output-file"));
+  }
+
+  std::ostream &out =
+    output_file ? (std::ostream &)file : (std::ostream &)message.result();
+
+  if(symtab_snapshot)
+  {
+    symbol_tablet snapshot = gdb_value_extractor.get_snapshot_as_symbol_table();
+    show_symbol_table(snapshot, ui_message_handler);
+  }
+  else
+  {
+    std::string snapshot = gdb_value_extractor.get_snapshot_as_c_code();
+    out << snapshot;
+  }
+
+  if(output_file)
+  {
+    file.close();
+  }
+  else
+  {
+    message.result() << messaget::eom;
+  }
+
+  return CPROVER_EXIT_SUCCESS;
+}
+
+void memory_analyzer_parse_optionst::help()
+{
+  message.status()
+    << '\n'
+    << banner_string("Memory-Analyzer", CBMC_VERSION) << '\n'
+    << align_center_with_border("Copyright (C) 2019") << '\n'
+    << align_center_with_border("Malte Mues, Diffblue Ltd.") << '\n'
+    << align_center_with_border("info@diffblue.com") << '\n'
+    << '\n'
+    << "Usage:                       Purpose:\n"
+    << '\n'
+    << " memory-analyzer [-?] [-h] [--help]  show help\n"
+    << " memory-analyzer --version           show version\n"
+    << " memory-analyzer <options> <binary>  analyze binary"
+    << "\n"
+    << " --core-file <file>           analyze from core file\n"
+    << " --breakpoint <breakpoint>    analyze from breakpoint\n"
+    << " --symbols <symbol-list>      list of symbols to analyze\n"
```

Comment:
Should be explicit that this is a required option

Reply:
Made explicit.